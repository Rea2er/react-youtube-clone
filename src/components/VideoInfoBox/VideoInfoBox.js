import React, { Component } from "react";
import { Image, Button } from "semantic-ui-react";
import "./VideoInfoBox.scss";

class VideoInfoBox extends Component {
  state = {
    collapsed: true,
  };

  onToggleCollapseButtonClick = () => {
    this.setState((prevState) => {
      return {
        collapsed: !prevState.collapsed,
      };
    });
  };

  render() {
    let descriptionTextClass = "collapsed";
    let buttonTitle = "Show More";
    if (!this.state.collapsed) {
      descriptionTextClass = "expanded";
      buttonTitle = "Show Less";
    }

    return (
      <div className="video-info-box">
        <Image
          className="channel-image"
          src="http://via.placeholder.com/48x48"
          circular
        />
        <div className="video-info">
          <div className="channel-name">Channel Name</div>
          <div className="video-publication-date">Thu 24, 2017</div>
        </div>
        <Button color="youtube">91.5K Subscribe</Button>
        <div className="video-description">
          <div className={descriptionTextClass}>
            <p>Paragraph 1</p>
            <p>Paragraph 2</p>
            <p>Paragraph 3</p>
            <p>Paragraph 4</p>
            <p>Paragraph 5</p>
          </div>
          <Button compact onClick={this.onToggleCollapseButtonClick}>
            {buttonTitle}
          </Button>
        </div>
      </div>
    );
  }
}

export default VideoInfoBox;
