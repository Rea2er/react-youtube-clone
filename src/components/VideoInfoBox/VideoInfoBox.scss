.video-info-box {
  display: grid;
  grid: auto auto / 58px auto max-content;
  align-items: center;
  grid-row-gap: 16px;
}
