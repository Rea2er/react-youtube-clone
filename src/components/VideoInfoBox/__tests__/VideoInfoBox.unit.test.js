import React from "react";
import { shallow } from "enzyme";
import VideoInfoBox from "../VideoInfoBox";

describe("VideoInfoBox", () => {
  test("renders collapsed", () => {
    const wrapper = shallow(<VideoInfoBox />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders expanded", () => {
    const wrapper = shallow(<VideoInfoBox />);
    wrapper.setState({ collapsed: false });
    expect(wrapper).toMatchSnapshot();
  });
});
