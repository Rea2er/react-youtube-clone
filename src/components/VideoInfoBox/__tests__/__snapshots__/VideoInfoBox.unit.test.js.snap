// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfoBox renders collapsed 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-publication-date"
    >
      Thu 24, 2017
    </div>
  </div>
  <Button
    as="button"
    color="youtube"
  >
    91.5K Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="collapsed"
    >
      <p>
        Paragraph 1
      </p>
      <p>
        Paragraph 2
      </p>
      <p>
        Paragraph 3
      </p>
      <p>
        Paragraph 4
      </p>
      <p>
        Paragraph 5
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox renders expanded 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-publication-date"
    >
      Thu 24, 2017
    </div>
  </div>
  <Button
    as="button"
    color="youtube"
  >
    91.5K Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="expanded"
    >
      <p>
        Paragraph 1
      </p>
      <p>
        Paragraph 2
      </p>
      <p>
        Paragraph 3
      </p>
      <p>
        Paragraph 4
      </p>
      <p>
        Paragraph 5
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;
